source/includes/monitoring/sdam.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+require __DIR__ . '/vendor/autoload.php';
+
+// start-mysubscriber
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $stream;
+    public function __construct($stream)
+    {
+        $this->stream = $stream;
+    }
+    public function serverOpening(MongoDB\Driver\Monitoring\ServerOpeningEvent $event): void
+    {
+        fwrite($this->stream, sprintf(
+            'Server opening on %s:%s%s',
+            $event->getHost(),
+            $event->getPort(),
+            PHP_EOL,
+        ));
+    }
+    public function serverClosed(MongoDB\Driver\Monitoring\ServerClosedEvent $event): void {}
+    public function serverChanged(MongoDB\Driver\Monitoring\ServerChangedEvent $event): void {}
+    public function serverHeartbeatFailed(MongoDB\Driver\Monitoring\ServerHeartbeatFailedEvent $event): void {}
+    public function serverHeartbeatStarted(MongoDB\Driver\Monitoring\ServerHeartbeatStartedEvent $event): void {}
+    public function serverHeartbeatSucceeded(MongoDB\Driver\Monitoring\ServerHeartbeatSucceededEvent $event): void {}
+    public function topologyChanged(MongoDB\Driver\Monitoring\TopologyChangedEvent $event): void {}
+    public function topologyClosed(MongoDB\Driver\Monitoring\TopologyClosedEvent $event): void {}
+    public function topologyOpening(MongoDB\Driver\Monitoring\TopologyOpeningEvent $event): void {}
+}
+// end-mysubscriber
+
+$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
+$client = new MongoDB\Client($uri);
+
+$database = $client->db;
+$collection = $database->my_coll;
```

Comment:
```suggestion
$collection = $client->db->my_coll;
```

`$database` is never used.